File: src/main/java/io/openfuture/openmessanger/repository/GroupParticipantRepository.java
package io.openfuture.openmessanger.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import io.openfuture.openmessanger.repository.entity.GroupChat;
import io.openfuture.openmessanger.repository.entity.GroupParticipant;

public interface GroupParticipantRepository extends JpaRepository<GroupParticipant, Integer> {
    Optional<GroupParticipant> findAllByParticipantAndGroupChat_Id(String username, Integer groupId);

    List<GroupParticipant> findAllByGroupChat(GroupChat groupChat);
}

File: src/main/kotlin/io/openfuture/openmessanger/repository/GroupParticipantRepository.kt
package io.openfuture.openmessanger.repository

import io.openfuture.openmessanger.repository.entity.GroupChat
import io.openfuture.openmessanger.repository.entity.GroupParticipant
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface GroupParticipantRepository : JpaRepository<GroupParticipant?, Int?> {
    fun findAllByParticipantAndGroupChat_Id(username: String?, groupId: Int?): Optional<GroupParticipant?>
    fun findAllByGroupChat(groupChat: GroupChat?): List<GroupParticipant?>
    fun findAllByGroupChatAndDeleted(groupChat: GroupChat?, deleted: Boolean): List<GroupParticipant?>
}